pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -61,13 +63,30 @@ makeWrapper() {
             command="${params[$((n + 1))]}"
             n=$((n + 1))
             echo "$command" >> "$wrapper"
-        elif [[ ("$p" == "--suffix") || ("$p" == "--prefix") ]]; then
+        elif [[ ("$p" == "--suffix") || ("$p" == "--prefix") || ("$p" == "--suffix-dedup") || ("$p" == "--prefix-dedup") ]]; then
             varName="${params[$((n + 1))]}"
             separator="${params[$((n + 2))]}"
             value="${params[$((n + 3))]}"
             n=$((n + 3))
             if test -n "$value"; then
-                if test "$p" = "--suffix"; then
+                if [[ "$p" == *'-dedup' ]]; then
+                    {
+                        # We need extended globbing for ?()
```

Comment:
Instead of changing `extglob`, what about copying `${separator}\${$varName}${separator}` (please add correct quoting) into a temporary unexported variable? Then you can write the second line without `?()`, so you can drop `shopt`, and presumably solve @talyz's second issue.